tests/testthat/test-cdpsession.R

```diff
@@ -83,3 +82,128 @@ test_that("connect() raises an error (or return a rejected promise) when there i
   expect_is(client_pr, "promise")
   expect_error(hold(client_pr))
 })
+
+test_that("Command-A command without a callback returns a promise whose value is a named list", {
+  client <- hold(chrome$connect())
+  pr <- client$Page$enable()
+  res <- hold(pr)
+  expect_is(res, "list")
+  expect_true(rlang::is_named(res))
+  hold(client$disconnect())
+})
+
+test_that("Command-The object passed to the callback function is identical to the value of the fulfilled promise", {
+  client <- hold(chrome$connect())
+  res_callback <- NULL
+  client$Page$getFrameTree(callback = function(res) {res_callback <<- res})
+  later::run_now(0.1)
+  res_pr <- hold(client$Page$getFrameTree())
+  expect_identical(res_callback, res_pr)
+  hold(client$disconnect())
+})
+
+test_that("Command-rlang lambda functions can be used in callbacks", {
+  client <- hold(chrome$connect())
+  res_native_fun <- NULL
+  res_rlang_fun <- NULL
+  client$Browser$getVersion(callback = function(res) {res_native_fun <<- res})
+  client$Browser$getVersion(callback = ~ {res_rlang_fun <<- .x})
+  hold(client$Browser$getVersion())
+  expect_identical(res_native_fun, res_rlang_fun)
+  hold(client$disconnect())
+})
+
+test_that("Event listener-When a callback is used, the returned function dismisses the listener", {
+  client <- hold(chrome$connect())
+  hold(client$Page$enable())
+  witness <- client$Page$loadEventFired()
+  expect_is(witness, "promise")
```

Comment:
```suggestion
  expect_s3_class(witness, "promise")
```
this is adviced instead of `expect_is`